pkg/directory/gh.go

```diff
@@ -0,0 +1,363 @@
+package directory
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// GHAllTeam containing all users in GitHub
+	GHAllTeam = "all"
+)
+
+// UsersService holds methods used in the GitHub UsersService for easier testing
+type UsersService interface {
+	Get(context.Context, string) (*github.User, *github.Response, error)
+}
+
+// GH hosts a client for accessing GH as well as cached Member and Team lists
+type GH struct {
+	*github.Client
+
+	UsersService UsersService
+	Info
+}
+
+// NewGitHub returns an initialized GitHub client to the caller and stored GH members and teams
+func NewGitHub(org string) (*GH, error) {
+	ctx := context.Background()
```

Comment:
Consider setting a timeout.

Reply:
I added a timeout in the individual requests handled inside of the library.

Reply:
👍